logutils/logutils_test.go

```diff
@@ -64,3 +71,241 @@ var _ = Describe("Logutils", func() {
 		Expect(buf.String()).To(ContainSubstring("logutils_test.go"))
 	})
 })
+
+var _ = DescribeTable("Formatter",
+	func(entry log.Entry, expectedLog, expectedSyslog string) {
+		f := &Formatter{}
+		out, err := f.Format(&entry)
+		Expect(err).NotTo(HaveOccurred())
+		expectedLog = strings.Replace(expectedLog, "<PID>", fmt.Sprintf("%v", os.Getpid()), 1)
+		Expect(string(out)).To(Equal(expectedLog))
+		expectedSyslog = strings.Replace(expectedSyslog, "<PID>", fmt.Sprintf("%v", os.Getpid()), 1)
+		Expect(FormatForSyslog(&entry)).To(Equal(expectedSyslog))
+	},
+	Entry("Empty", log.Entry{},
+		"0001-01-01 00:00:00.000 [PANIC][<PID>] <nil> <nil>: \n",
+		"PANIC <nil> <nil>: \n"),
+	Entry("Basic",
+		log.Entry{
+			Level: log.InfoLevel,
+			Time:  theTime(),
+			Data: log.Fields{
+				"__file__": "foo.go",
+				"__line__": 123,
+			},
+			Message: "The answer is 42.",
+		},
+		"2017-03-15 11:22:33.123 [INFO][<PID>] foo.go 123: The answer is 42.\n",
+		"INFO foo.go 123: The answer is 42.\n",
+	),
+	Entry("With fields",
+		log.Entry{
+			Level: log.WarnLevel,
+			Time:  theTime(),
+			Data: log.Fields{
+				"__file__": "foo.go",
+				"__line__": 123,
+				"a":        10,
+				"b":        "foobar",
+				"c":        theTime(),
+				"err":      errors.New("an error"),
+			},
+			Message: "The answer is 42.",
+		},
+		"2017-03-15 11:22:33.123 [WARNING][<PID>] foo.go 123: The answer is 42. a=10 b=\"foobar\" c=2017-03-15 11:22:33.123 +0000 UTC err=an error\n",
+		"WARNING foo.go 123: The answer is 42. a=10 b=\"foobar\" c=2017-03-15 11:22:33.123 +0000 UTC err=an error\n"),
+)
+
+func theTime() time.Time {
+	theTime, err := time.Parse("2006-01-02 15:04:05.000", "2017-03-15 11:22:33.123")
+	if err != nil {
+		panic(err)
+	}
+	return theTime
+}
+
+var _ = Describe("StreamDestination", func() {
+	var s *StreamDestination
+	var pr *io.PipeReader
+	var pw *io.PipeWriter
+	var c chan<- QueuedLog
+	var mt *mockMonotime
+
+	BeforeEach(func() {
+		pr, pw = io.Pipe()
+		mt = &mockMonotime{
+			time: 100 * time.Hour,
+		}
+		s = NewStreamDestination(log.InfoLevel, pw, mt)
+		go s.LoopWritingLogs()
+		c = s.Channel()
+	})
+	AfterEach(func() {
+		close(c)
+	})
+
+	It("should write a log (no WaitGroup)", func() {
+		c <- QueuedLog{
+			Level:         log.InfoLevel,
+			Message:       []byte("Message"),
+			SyslogMessage: "syslog message",
+		}
+		b := make([]byte, 1024)
+		n, err := pr.Read(b)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(string(b[:n])).To(Equal("Message"))
+	})
+
+	It("should log dropped logs after time advances far enough", func() {
```

Comment:
Actually, why don't I just simplify the dropped logs logging and remove the time element.